user/installing-dependencies.md

```diff
@@ -61,21 +61,33 @@ Say your project requires the pngquant tool to compress PNG files, here's how to
 
 You can also use the APT addon.
 
+This addon provides declarative shortcuts to basic operations of the `apt-get` commands.
+
+If your requirements goes beyond the normal installation, please use another method described above.
+
 #### Adding APT Sources
 
-To add APT sources from the [source whitelist](https://github.com/travis-ci/apt-source-whitelist) before your custom build steps, use the `addons.apt.sources` key:
+To add APT sources, you can use one of the following three types of entries:
+
+1. aliases defined in [source whitelist](https://github.com/travis-ci/apt-source-whitelist)
+1. `sourceline` key-value pairs which will be added to `/etc/apt/sources.list`
+1. when APT sources require GPG keys, you can specify this with `key_url` pairs in addition to `sourceline`.
+
+The following snippet shows these three types of APT sources
 
 ``` yaml
 addons:
   apt:
     sources:
     - deadsnakes
-    - ubuntu-toolchain-r-test
+    - sourceline: 'ppa:ubuntu-toolchain-r/test'
+    - sourceline: 'deb https://packagecloud.io/chef/stable/ubuntu/precise main'
+      key_url: 'https://packagecloud.io/gpg.key'
 ```
 
 #### Adding APT Packages
 
-To install packages from the [package whitelist](https://github.com/travis-ci/apt-package-whitelist)  before your custom build steps, use the `addons.apt.packages` key:
+List APT packages under the `addons.apt.packages` key:
```

Comment:
This is the part regarding the GCE builds, where `sudo` is available, so we are simply ignoring the whitelist and installing any package the user specifies in the list here.

(There is a separate section on the container builds below that still mentions the whitelist.)